docs/core-concepts/authentication/sdk.mdx

```diff
@@ -1,230 +1,397 @@
   email: 'user@example.com',
   password: 'secure_password123'
 })
+```
 
-if (data) {
-  // Token is automatically stored and used for future requests
-  console.log('Logged in:', data.user)
-  console.log('Token:', data.accessToken)
+### Output
+
+```json
+{
+  "data": {
+    "user": {
+      "id": "usr_abc123",
+      "email": "user@example.com",
+      "name": "John Doe",
+      "emailVerified": true,
+      "createdAt": "2024-01-15T10:00:00Z",
+      "updatedAt": "2024-01-15T10:00:00Z"
+    },
+    "accessToken": "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9..."
+  },
+  "error": null
 }
 ```
 
-### Sign Out
+---
 
-```javascript
-const { error } = await insforge.auth.signOut()
-// Clears stored tokens and session
+## signInWithOAuth()
+
+Start OAuth authentication flow with Google or GitHub.
+
+### Parameters
+
+- `provider` ('google' | 'github', required) - OAuth provider
+- `redirectTo` (string, optional) - URL to redirect after authentication
+- `skipBrowserRedirect` (boolean, optional) - Return URL instead of auto-redirecting
+
+### Returns
+
+```typescript
+{
+  data: { url?: string, provider?: string },
+  error: Error | null
+}
 ```
 
-## OAuth Authentication
+<Note>
+After OAuth redirect, SDK automatically detects callback parameters and saves session. No manual handling needed.
+</Note>
 
-The SDK includes automatic OAuth callback detection (v0.0.14+). When users return from OAuth providers, the SDK automatically handles the authentication.
+### Example
 
-<CodeGroup>
-```javascript Google OAuth
-// Redirects to Google OAuth
-await insforge.auth.signInWithOAuth({
+```javascript
+const { data, error } = await insforge.auth.signInWithOAuth({
   provider: 'google',
   redirectTo: 'http://localhost:3000/dashboard'
 })
 
-// SDK automatically detects and handles the OAuth callback
+// User is redirected to Google OAuth
+// After authentication, user returns to redirectTo URL
+// SDK automatically handles the callback and saves session
 ```
 
-```javascript GitHub OAuth
-// Redirects to GitHub OAuth
-await insforge.auth.signInWithOAuth({
-  provider: 'github',
-  redirectTo: 'http://localhost:3000/dashboard'
-})
+### Output
+
+```json
+{
+  "data": {
+    "url": "https://accounts.google.com/o/oauth2/v2/auth?client_id=...",
+    "provider": "google"
+  },
+  "error": null
+}
 ```
 
-```javascript Get URL Only
-// Returns URL without redirecting
-const { data } = await insforge.auth.signInWithOAuth({
-  provider: 'google',
-  skipBrowserRedirect: true
-})
+---
+
+## signOut()
 
-console.log('OAuth URL:', data.url)
+Sign out the current user and clear session.
+
+### Parameters
+
+None
+
+### Returns
+
+```typescript
+{
+  error: Error | null
+}
 ```
-</CodeGroup>
 
-<Note>
-OAuth requires environment variables to be configured on the backend:
-- `GOOGLE_CLIENT_ID`, `GOOGLE_CLIENT_SECRET`
-- `GITHUB_CLIENT_ID`, `GITHUB_CLIENT_SECRET`
-</Note>
+### Example
+
+```javascript
+const { error } = await insforge.auth.signOut()
```

Comment:
why does error exist, how to handle it? signOut failed and stay in loggedin?
If  signout always successes, the result isn't necessary.